openedx/core/djangoapps/theming/apps.py

```diff
@@ -0,0 +1,82 @@
+
+import os
+from django.apps import AppConfig
+from django.conf import settings
+from django.core.checks import Error, Tags, register
+
+
+class ThemingConfig(AppConfig):
+    name = 'openedx.core.djangoapps.theming'
+    verbose_name = "Theming"
+
+
+@register(Tags.compatibility)
+def check_comprehensive_theme_settings(app_configs, **kwargs):
+    """
+    Checks the comprehensive theming theme directory settings.
+
+    Raises compatibility Errors upon:
+        - COMPREHENSIVE_THEME_DIRS is not a list
+        - theme dir path is not a string
+        - theme dir path is not an absolute path
+        - path specified in COMPREHENSIVE_THEME_DIRS does not exist
+
+    Returns:
+        List of any Errors.
+    """
+    if not getattr(settings, "ENABLE_COMPREHENSIVE_THEMING", False):
```

Comment:
Doesn't this setting have to actually exist?  Defaulting to `False` could mask its absence, although I think other errors would get thrown before this point while initializing settings in that case.

Reply:
Good point - it should definitely exist. I'll remove this default so an error will be thrown if not.